__tests__/app.test.js

```diff
@@ -133,3 +133,98 @@ describe("/api/articles/:article_id/comments", () => {
       .then(({ body: { msg } }) => expect(msg).toBe("bad request"));
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("when given an object with an username and body it checks if the article exists and then adds the comment to the comments table", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { comment } }) =>
+        expect.objectContaining({
+          author: expect(newComment.username),
+          body: expect(newComment.body),
+          votes: expect(0),
+          article_id: expect(1),
+        })
+      );
+  });
+
+  test("when given a article_id that doesn't exist return a 'not-found' message", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/299/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("not found");
+      });
+  });
+
+  test("should return a sql error if the username doesn't exist in the users table ", () => {
+    const newComment = {
+      username: "vaish",
+      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+});
+
+describe.only("PATCH /api/article/:article_id", () => {
+  test("When given an object with the keys inc_votes does it increase/decrease the votes and return a status 200", () => {
+    const voteObj = { inc_votes: 10 };
+    const updatedArticle = {
+      title: "Moustache",
+      topic: "mitch",
+      author: "butter_bridge",
+      body: "Have you seen the size of that thing?",
+      created_at: 1602419040000,
+      votes: 10,
+    };
+    return request(app)
+      .patch("/api/articles/12")
+      .send(voteObj)
+      .expect(200)
+      .then(({ body: { article } }) => {
+        expect.objectContaining({
+          title: expect("Moustache"),
+          topic: expect("mitch"),
+          author: expect("butter_bridge"),
+          body: expect("Have you seen the size of that thing?"),
+          votes: expect(10),
+        });
+      });
+  });
+
+  test("when given an article that doesn't exist then it returns a 404", () => {
+    const voteObj = { inc_votes: 10 };
+    const updatedArticle = {
+      title: "Moustache",
+      topic: "mitch",
+      author: "butter_bridge",
+      body: "Have you seen the size of that thing?",
+      created_at: 1602419040000,
+      votes: 10,
+    };
+    return request(app)
+      .patch("/api/articles/100")
+      .send(voteObj)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("not found");
+      });
+  });
+});
```

Comment:
Hey Vaish. Nice tests so far, but could you just add two 400s? One for invalid ID (i.e. banana) and one for missing/incorrect body (i.e. inc_votes is not a number or is missing). 🙂